lib/stud/interval.rb

```diff
@@ -1,38 +1,83 @@
-require "stud/task"
+require "thread"
+
 module Stud
+  STUD_STOP_REQUESTED = :stud_stop_requested
+
+  extend self
```

Comment:
good concern. I usually use `extend self` for making "singleton" modules and to avoid having to define all methods using `self.` it makes refactoring such modules into proper classes easier. but it is true that it make the methods available to a class including it. 

no strong feeling. I can definitely revert back to no `extend self` plus its not really in the scope of this PR.